js-watcher/server.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+type Server struct {
+	caches CacheCollection
+	config Config
+	router *httprouter.Router
+	server *http.Server
```

Comment:
A server inside a server looks v strange to me; also, having the handler as some kind of parent struct really give off some kind of code smell to me; however, I don't know exactly what is the better approach rn...

Reply:
Embedding `htts.Server` as I suggested above works. And it removes all cases of `server.server` weirdness except when we construct a new server. But I think this is better than what I had previously.